### PR TITLE
buildGoModule: correct spelling in documentation and simplify the go-modules structure

### PR DESCRIPTION
**Pull request by ShamrockLee**
###### Description of changes

Change "platform dependant checksums" to "platform-dependent checksums", as "dependant" is only used as a noun.

Simplify the attribute structure in `pkgs/build-support/go/modules.nix`. This causes no rebuild and ease the review process of #225051.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by zowoq** [comment]
Switching to draft as we'll merge https://github.com/NixOS/nixpkgs/pull/225371 first.

**Comment by zowoq** [comment]
@ShamrockLee Please rebase this PR.

File: pkgs/build-support/go/module.nix [hunk 1]
```diff
@@ -2,63 +2,65 @@
...
+}@args':
 
 assert goPackagePath != "" -> throw "`goPackagePath` is not needed with `buildGoModule`";
 assert (args' ? vendorHash && args' ? vendorSha256) -> throw "both `vendorHash` and `vendorSha256` set. only one can be set.";
 
 let
   args = removeAttrs args' [ "overrideModAttrs" "vendorSha256" "vendorHash" ];
 
-  go-modules = if (vendorHash != null) then stdenv.mkDerivation (let modArgs = {
+  go-modules = if (isNull vendorHash) then "" else
```
Comment (zowoq) [inline pkgs/build-support/go/module.nix:58]: > https://nixos.org/manual/nix/stable/language/builtins.html#builtins-isNull
> This function is deprecated; just write e == null instead.

**Comment by zowoq** [comment]
Please revert the last indentation change.

File: pkgs/build-support/go/module.nix [hunk 2]
```diff
@@ -2,149 +2,152 @@
...
-
-    configurePhase = args.modConfigurePhase or ''
-      runHook preConfigure
-      export GOCACHE=$TMPDIR/go-cache
-      export GOPATH="$TMPDIR/go"
-      cd "${modRoot}"
-      runHook postConfigure
-    '';
+  go-modules =
+    if (isNull vendorHash) then "" else
```
Comment (zowoq) [inline pkgs/build-support/go/module.nix:59]: Sorry, in my previous comment I meant revert this:
```suggestion
  go-modules = if (isNull vendorHash) then "" else
```

File: pkgs/build-support/go/module.nix (hunk 1 above)
Comment (ShamrockLee) [reply pkgs/build-support/go/module.nix]: Addressed.

**Comment by zowoq** [comment]
@ShamrockLee Please drop the formatting changes from this PR, rather than us going back and forth over formatting it'll be easier if I do it myself.

**Comment by ShamrockLee** [comment]
Finished rebasing and addressed the suggestions.

**Pull request merged**